### PR TITLE
chore(deps): Bump Rust to 1.77.0

### PR DESCRIPTION
**Pull request by bruceg**
Please pay special attention to the code deletions here. The compiler identified several unused members of structs, and the fixups I made may not be entirely correct.

### CODE REVIEW

File: src/sinks/aws_s3/integration_tests.rs [hunk 1]
```diff
@@ -388,7 +388,11 @@ async fn s3_healthchecks() {
         .create_service(&ProxyConfig::from_env())
         .await
         .unwrap();
-    config.build_healthcheck(service.client()).unwrap();
+    config
+        .build_healthcheck(service.client())
+        .unwrap()
+        .await
```
Comment (tobz) [inline src/sinks/aws_s3/integration_tests.rs:394]: Huh, should we have always been awaiting this? 😅

**Review by tobz — APPROVED** [review]
Overall, this looks good to me.

Thinking about each change, it seems pretty clear that it's semantically valid. My main thought is whether or not anything depends on those now-removed fields not dropping as early as they're now going to.

**Comment by bruceg** [comment]
> My main thought is whether or not anything depends on those now-removed fields not dropping as early as they're now going to.

Yes, I wondered about that too, particularly about the channel, but couldn't determine if it mattered.

**Comment by tobz** [comment]
> Yes, I wondered about that too, particularly about the channel, but couldn't determine if it mattered.

If the tests pass, at least, then we're probably fine.

**Comment by neuronull** [comment]
Seems to have introduced a failure in the journald unit tests around checkpointing...

**Comment by bruceg** [comment]
There are test failures happening from an inability to run `protoc`:
* https://github.com/vectordotdev/vector/actions/runs/8382354304/job/22956027758?pr=20149#step:6:2462
* https://github.com/vectordotdev/vector/actions/runs/8382354307/job/22956011512?pr=20149#step:6:2809

Did something change with that recently? I don't think this PR would impact that.

**Comment by bruceg** [comment]
Apparently, all the [current Docker images for 1.77](https://hub.docker.com/_/rust/tags?page=1&name=1.77) are only `linux/i386` arch, so this is broken by outside forces for now.

**Pull request merged**